lib/iris/experimental/ugrid.py

```diff
@@ -998,97 +1030,614 @@ def equal(self, other, lenient=None):
-#         self.edge_x = edge_x
-#         self.edge_y = edge_y
-#
-#         # WOO-GA - this can easily get out of sync with the self attributes.
-#         # choose the container wisely e.g., could be an dict..., also the self
-#         # attributes may need to be @property's that access the chosen _members container
-#         self._members = [ ... ]
-#
-#     def __iter__(self):
-#         for member in self._members:
-#             yield member
-#
-#     def __getstate__(self):
-#         pass
-#
-#     def __setstate__(self, state):
-#         pass
-#
-#     def coord(self, **kwargs):
-#         # see Cube.coord for pattern, checking for a single result
-#         return self.coords(**kwargs)[0]
-#
-#     def coords(self, ...):
-#         # see Cube.coords for relevant patterns
-#         # return [ ... ]
-#         pass
-#
-#     def add(self, **kwargs):
-#         pass
-#
-#     def remove(self, ...):
-#         # needs to respect the minimum UGRID contract
-#         # use logging/warning to flag items not removed - highlight in doc-string
-#         # don't raise an exception
-#
-#     def __str__(self):
-#         pass
-#
-#     def __repr__(self):
-#         pass
-#
-#     def __eq__(self, other):
-#         # Full equality could be MASSIVE, so we want to avoid that.
-#         # Ideally we want a mesh signature from LFRic for comparison, although this would
-#         # limit Iris' relevance outside MO.
-#         # TL;DR: unknown quantity.
-#         raise NotImplemented
-#
-#     def __ne__(self, other):
-#         # See __eq__
-#         raise NotImplemented
-#
-#
-# class _Mesh2DCoordinateManager(_Mesh1DCoordinateManager):
-#     OPTIONAL = (
-#         "edge_x",
-#         "edge_y",
-#         "face_x",
-#         "face_y",
-#     )
-#     def __init__(self, node_x, node_y, edge_x=None, edge_y=None, face_x=None, face_y=None):
-#         # optional
-#         self.face_x = face_x
-#         self.face_y = face_y
-#
-#         super().__init__(node_x, node_y, edge_x=edge_x, edge_y=edge_y)
-#
-#         # does the order matter?
-#         self._members.extend([self.face_x, self.face_y])
-#
 #
+
+
+class _Mesh1DCoordinateManager:
+    """
+
+    TBD: require clarity on coord_systems validation
+    TBD: require clarity on __eq__ support
+    TBD: rationalise self.coords() logic with other manager and Cube
+
+    """
+
+    REQUIRED = (
+        "node_x",
+        "node_y",
+    )
+    OPTIONAL = (
+        "edge_x",
+        "edge_y",
+    )
+
+    def __init__(self, node_x, node_y, edge_x=None, edge_y=None):
+        # initialise all the coordinates
+        self.ALL = self.REQUIRED + self.OPTIONAL
+        self._members = {member: None for member in self.ALL}
+
+        # required coordinates
+        self.node_x = node_x
+        self.node_y = node_y
+        # optional coordinates
+        self.edge_x = edge_x
+        self.edge_y = edge_y
+
+    def __eq__(self, other):
+        # TBD
+        raise NotImplementedError
```

Comment:
I've switched to `return NotImplemented`, which has appeased my linter - I don't think it liked me raising a `NotImplementedError` in an `ABC` then not overriding.

Reply:
My mistake, this should be `return NotImplemented` but you `raise NotImplementedError` :+1: